### PR TITLE
Initial support for Rust auto-indexing

### PR DESCRIPTION
**Pull request by varungandhi-src**
See also: https://github.com/sourcegraph/lsif-rust/pull/1

### CODE REVIEW

**Review by olafurpg — COMMENTED** [review]
Looks great!

File: lib/codeintel/autoindex/inference/go_test.go [hunk 1]
```diff
@@ -10,36 +9,14 @@ import (
 )
 
 func TestGoPatterns(t *testing.T) {
-	testCases := []struct {
-		path     string
-		expected bool
-	}{
+	implTestLangPatterns(t, GoPatterns(), []PathTestCase{
```
Comment (efritz) [inline lib/codeintel/autoindex/inference/go_test.go:12]: `TestJavaPatterns` could also be simplified

File: lib/codeintel/autoindex/inference/lang_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"testing"
+)
+
+type PathTestCase = struct {
+	path     string
+	expected bool
+}
+
+// Test if all test cases match against any pattern in patterns.
+func implTestLangPatterns(t *testing.T, patterns []*regexp.Regexp, testCases []PathTestCase) {
```
Comment (efritz) [inline lib/codeintel/autoindex/inference/lang_test.go:16]: Can we rename this to simply `testLangPatterns`?

File: lib/codeintel/autoindex/inference/lang_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package inference
```
Comment (efritz) [inline lib/codeintel/autoindex/inference/lang_test.go:1]: Idiomatically, we'd call this file `helpers_test.go`.

**Review by efritz — APPROVED** [review]

**Comment by varungandhi-src** [comment]
Thanks for the comments, I wasn't sure about the naming conventions, so I figured I'd put something up. :) Fixed the code and refactored things out a bit more.

**Pull request merged**